### PR TITLE
Added option to directly edit the project title from header

### LINKED ISSUE #1808: Add feature to directly edit the project title
**Is your feature request related to a problem? Please describe.**
Right now, a prompt opens up asking for the title, we can improve this experience if we could directly edit the project title by clicking on it.

**Describe the solution you'd like**
Edit the project title by directly clicking on the title, it can be implemented by using an input tag in place of the span tag 

**Screenshots**
image

### PR DESCRIPTION
**Pull request by dipanshparmar**
Fixes #1808 

#### Describe the changes you have made in this PR -
1. Added option to directly edit the project title from the header itself.

### Screenshots of the changes (If any) -
**Initial State**
image

**On Change**
(Changes are synced automatically between both of the project title inputs)
image

### CODE REVIEW

File: public/src/ux.js [hunk 1]
```diff
@@ -502,20 +502,24 @@ $('#octalInput').on('keyup', () => {
     setBaseValues(x);
 });
 
+// Function updates the values of project inputs accordingly
+const watchOneInputChangeAndUpdateAnother = (objectToListen, objectToChange) => $(`${objectToListen}`).on('input', (e) => { $(`${objectToChange}`).val(e.target.value); });
```
Comment (satu0king) [inline public/src/ux.js:506]: Neat!

File: public/src/ux.js [hunk 2]
```diff
@@ -502,20 +502,24 @@ $('#octalInput').on('keyup', () => {
...
     setupPanelListeners('#layoutDialog');
     setupPanelListeners('#verilogEditorPanel');
 
-    $('#projectName').on('click', () => {
-        $("input[name='setProjectName']").focus().select();
-    });
+    // setting the name of the title bar as same as it is in the side bar
+    $('.projectName').val($('[name="setProjectName"]').val()); // setting the initial value of title heading
+    watchOneInputChangeAndUpdateAnother('[name="setProjectName"]', '.projectName'); // setting the value of header title on sidebar title change
+    watchOneInputChangeAndUpdateAnother('.projectName', '[name="setProjectName"]'); // setting the value of sidebar title on header title change
```
Comment (satu0king) [inline public/src/ux.js:522]: You actually made a mistake. The name of the project itself is not changing, only the UI is changing, please fix this.

File: public/src/listeners.js [hunk 3]
```diff
@@ -55,11 +55,16 @@ export default function startListeners() {
...
+    // $('#exitViewBtn').click(() => showAll());
+    window.addEventListener('keyup', (e) => {
+        scheduleUpdate(1);
+        simulationArea.shiftDown = e.shiftKey;
+        if (e.keyCode == 16) {
+            simulationArea.shiftDown = false;
+        }
+        if (e.key == 'Meta' || e.key == 'Control') {
+            simulationArea.controlDown = false;
+        }
```
Comment (satu0king) [inline public/src/listeners.js:67]: I think this is a mistake, please remove this.

**Review by satu0king — CHANGES_REQUESTED** [review]
Works nicely but please fix conflicts and the changes requested.

**Comment by dipanshparmar** [comment]
> Works nicely but please fix conflicts and the changes requested.

@satu0king working on this!

**Comment by dipanshparmar** [comment]
@satu0king please have a look now.

**Comment by dipanshparmar** [comment]
@satu0king closing the PR as it is related to new_UI branch and making a new one for master.

**Pull request closed without merging**